Tomlet/TomlParser.cs

```diff
@@ -805,112 +811,61 @@ private void ReadTableArrayStatement(StringReader reader, TomlDocument document)
             if (!reader.ExpectAndConsume(']') || !reader.ExpectAndConsume(']'))
                 throw new UnterminatedTomlTableArrayException(_lineNumber);
 
-            TomlTable parentTable;
-            if (_lastTableArrayName != null && arrayName.StartsWith(_lastTableArrayName + "."))
-            {
-                //nested array of tables directly relative to parent - we can cheat
-
-                //Save parent table
-                parentTable = _currentTable!;
+            TomlTable parentTable = document;
+            var relativeKey = arrayName;
+            FindParentAndRelativeKey(ref parentTable, ref relativeKey);
 
-                //Work out relative key
-                var relativeKey = arrayName.Replace(_lastTableArrayName + ".", "");
-
-                //Make new array and table
-                var newArray = new TomlArray();
-                _currentTable = new TomlTable();
-                newArray.ArrayValues.Add(_currentTable);
-
-                //Insert into parent table
-                parentTable.ParserPutValue(relativeKey, newArray, _lineNumber);
-
-                //Save variables
-                _lastTableArrayName = arrayName;
-                return;
-            }
-
-            if (TomlKeyUtils.IsSimpleKey(arrayName))
+            //Find existing array or make new one
+            TomlArray array;
+            if (parentTable.ContainsKey(relativeKey))
             {
-                //Not present - create and populate with one table.
-                _currentTable = new TomlTable();
-
-                TomlArray tableArray;
-
-                //Simple key so looking up via document.ContainsKey is fine.
-                if (!document.ContainsKey(arrayName))
-                    //make a new one if it doesn't exist
-                    tableArray = new TomlArray {IsLockedToBeTableArray = true};
-                else if (document.Entries.TryGetValue(arrayName, out var hopefullyArray) && hopefullyArray is TomlArray arr)
-                    //already exists, use it
-                    tableArray = arr;
-                else
-                    throw new TomlTableArrayAlreadyExistsAsNonArrayException(_lineNumber, arrayName);
-
-                if (!tableArray.IsLockedToBeTableArray)
+                array = parentTable.GetArray(relativeKey);
```

Comment:
If possible, can you change this to `var value = parentTable.GetValue(relativeKey)`, then check 
```cs
if(value is TomlArray arr)
	array = arr;
else
	throw new TomlTableArrayAlreadyExistsAsNonArrayException(_lineNumber, arrayName);
```
That way we preserve the original behavior.